lib/SILOptimizer/Mandatory/TransferNonSendable.cpp

```diff
@@ -530,16 +530,21 @@ class PartitionOpTranslator {
   /// NOTE: We special case RawPointer and NativeObject to ensure they are
   /// treated as non-Sendable and strict checking is applied to it.
   bool isNonSendableType(SILType type) const {
-    switch (type.getASTType()->getKind()) {
-    case TypeKind::BuiltinNativeObject:
-    case TypeKind::BuiltinRawPointer:
-      // These are very unsafe... definitely not Sendable.
+    // Treat Builtin.NativeObject and Builtin.RawPointer as non-Sendable.
+    if (type.getASTType()->is<BuiltinNativeObjectType>() ||
+        type.getASTType()->is<BuiltinRawPointerType>()) {
       return true;
-    default:
-      // Consider caching this if it's a performance bottleneck.
-      return type.conformsToProtocol(function, sendableProtocol)
-          .hasMissingConformance(function->getParentModule());
     }
+
+    // Function types do not conform to protocols... so we need to check for
+    // Sendability ourselves.
+    if (auto *fas = type.getASTType()->getAs<SILFunctionType>()) {
+      return !fas->isSendable();
+    }
+
+    // Consider caching this if it's a performance bottleneck.
+    return type.conformsToProtocol(function, sendableProtocol)
+        .hasMissingConformance(function->getParentModule());
```

Comment:
Could you use `isSendableType` here instead?

Reply:
I added a method on SILType that calls that. I changed it to use this.